pdm/init/action.yml

```diff
@@ -54,10 +54,10 @@ runs:
     - name: Install dependencies (main, all extras and specified dev group)
       run: |
         echo "${{ github.workspace }}/.venv/bin/python" > .pdm-python
-        params="--group :all --dev"
-        [ -z "${GROUPS}" ] && params+=" --group ${GROUPS}"
-        [ -z "${EXCLUDED_GROUPS}" ] && params+=" --without ${EXCLUDED_GROUPS}"
-        pdm sync "${params}"
+        params=(--group :all --dev)
+        [ -n "${GROUPS}" ] && params+=(--group ${GROUPS})
```

Comment:
I tend to prefer using `[ -z ... ] || ...` or `if [ -n ...] ; then ... ; fi` for readability, and because using `[ -n ... ] && ...` returns an error code which is incompatible with bash mode `-e` (where when a command fails, the whole script fails)

Reply:
PR updated 👍🏼